src/sql/src/pure.rs

```diff
@@ -1310,29 +1284,27 @@ async fn purify_create_source(
 /// we are permitted to use async code.
 async fn purify_alter_source(
     catalog: impl SessionCatalog,
-    mut stmt: AlterSourceStatement<Aug>,
+    stmt: AlterSourceStatement<Aug>,
     storage_configuration: &StorageConfiguration,
-) -> Result<
-    (
-        Vec<(GlobalId, CreateSubsourceStatement<Aug>)>,
-        Statement<Aug>,
-    ),
-    PlanError,
-> {
+) -> Result<Vec<Statement<Aug>>, PlanError> {
     let scx = StatementContext::new(None, &catalog);
     let AlterSourceStatement {
-        source_name,
+        source_name: outer_source_name,
         action,
         if_exists,
-    } = &mut stmt;
+    } = stmt;
 
     // Get connection
-    let pg_source_connection = {
+    let (source_name, pg_source_connection, mut current_references) = {
```

Comment:
nit: there's a baked-in assumption here that this has to be a PG source -- despite the `purify_alter_source` method potentially applying to any source that has subsources

I can refactor this when I implement alter source for mysql, but since we're already changing a lot here it could be nice to rename `pg_source_connection` to `source_connection` and make things more generic below until we get to the pg-specific stuff around replication-slots, publications, etc (similar to `purify_create_source`)